Neo4j.Driver/Neo4j.Driver/Internal/IO/SpanPackStreamReader.cs

```diff
@@ -0,0 +1,503 @@
+// Copyright (c) "Neo4j"
+// Neo4j Sweden AB [http://neo4j.com]
+// 
+// This file is part of Neo4j.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using Neo4j.Driver.Internal.Messaging;
+
+namespace Neo4j.Driver.Internal.IO;
+
+internal ref struct SpanPackStreamReader
```

Comment:
I'm unsure why this has to be a value type and all the duplicate code that creates rather than using a class and shared interface with PackStreamReader.